toolbox/include/base/WBBlock.h

```diff
@@ -0,0 +1,79 @@
+#ifndef WBT_WBIBLOCK_H
+#define WBT_WBIBLOCK_H
+
+#include "Block.h"
+#include <string>
+#include <memory>
+#include <iDynTree/Core/VectorFixSize.h>
+
+namespace wbt {
+    class WBBlock;
+    class Configuration;
+    class BlockInformation;
+    class RobotInterface;
+}
+
+namespace iDynTree {
+    class Twist;
+    class Transform;
+    class MatrixDynSize;
+    class VectorDynSize;
+}
+
+/**
+ * \struct iDynTreeRobotState WBBlock.h
+ *
+ * This struct contains the pointer to iDynTree objects used to configure the
+ * state of iDynTree::KinDynComputations objects.
+ */
+struct iDynTreeRobotState {
+    iDynTree::Twist*         m_baseVelocity   = nullptr;
+    iDynTree::Vector3*       m_gravity        = nullptr;
+    iDynTree::Transform*     m_world_T_base   = nullptr;
+    iDynTree::VectorDynSize* m_jointsVelocity = nullptr;
+    iDynTree::VectorDynSize* m_jointsPosition = nullptr;
+};
+
+/**
+ * TODO
+ * Basic class for WBI related blocks.
+ * This class (the whole toolbox in reality) assumes the block represent
+ * an instantaneous system (i.e. not a dynamic system).
+ *
+ * You can create a new block by deriving this class and implementing at least
+ * the output method.
+ *
+ * This block implements the following default behaviours:
+ * - it ask for 4 parameters (robot name, local (module) name, names of the remote control boards,
+ *   and the list of joints which should all belong to one of the remote control boards)
+ * - It initializes the yarp network and the whole body interface object
+ * - During terminate it closes and release the interface object and terminate the yarp network
+ *
+ * @note Usually you want to call this class implementations at some point in your
+ * method overridings, unless you want to completely change the code (but at that point
+ * you probabily want to derive from Block instead)
+ */
+class wbt::WBBlock : public wbt::Block
+{
+private:
+    static const unsigned PARAM_IDX_CONFIGURATION;
+    static const unsigned PARAM_IDX_CONFBLOCKNAME;
+
+protected:
+    std::string confKey;
```

Comment:
`configurationID` ?

Reply:
This is the key of the map contained in `ToolboxSingleton`, `ID` seems too generic in my opinion. The block itself should not be aware of this key, and gathering the correct information is handled directly from its `WBBlock` parent